cmake/onnxruntime_unittests.cmake

```diff
@@ -1553,7 +1553,7 @@ if (NOT CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
     endif()
   endif()
 
-  if (onnxruntime_BUILD_SHARED_LIB AND (NOT onnxruntime_MINIMAL_BUILD OR onnxruntime_MINIMAL_BUILD_CUSTOM_OPS))
+  if (onnxruntime_BUILD_SHARED_LIB AND (NOT onnxruntime_MINIMAL_BUILD OR onnxruntime_MINIMAL_BUILD_CUSTOM_OPS) AND NOT CMAKE_SYSTEM_NAME STREQUAL "Android")
```

Comment:
`NOT ANDROID` is more concise
https://cmake.org/cmake/help/latest/variable/ANDROID.html